WF_Proton_Launcher/Main.cpp

```diff
@@ -119,6 +120,8 @@ int main(int argc, char** argv)
 			verbose = true;
 		if (arg == "-32")
 			warframe_exe = WARFRAME_EXE;
+		if (arg == "-registry:Steam")
```

Comment:
It's probably a better idea to do a substring match on here, just check if the start of the argument is "-registry:".

Reply:
you could, although that does leave it open for misinterpretation if they pass another registry option later (as far as I know Steam is the only one)

Reply:
On the other hand, doing it as a substring match would be a more future-proof change, in case DE decides to add another registry instead of just Steam, and someone wants to use this launcher outside of Steam (which is a reasonable thing to do for running Warframe with regular Wine).  
It would also not hide the error in case someone were to misspell Steam when writing the launch option manually.

Reply:
how about something like:
```
		if (arg.substr(0,10) == "-registry:")
                {
                    switch(arg.substr(11)){
                        case "Steam" :
                            REGISTRY_STEAM = " " + arg;
                        default :
                            REGISTRY_STEAM = " " + arg;
                    }
                }
```

Reply:
I'm just curious why that would even be necessary? Warframe is after all a much better judge at what a valid registry option is, so doing explicit handling for it in the launcher will at best just cause issues in the future if they want to change or add additional ones.

Reply:
it's not necessary, just an additional method of parsing registry keys. I suppose just checking for -registry is enough.

Reply:
how about just this:
```
bool do_update = true,
     do_cache = false,
     do_launch = true,
     redownload = false,
     verbose = false,
     steam = false;
```
and
```
		if (arg.substr(0,9) == "-registry")
                {
                    REGISTRY_STEAM = " -registry:Steam";
                    steam = true;
                }
```
this way we can detect if it's a steam version and change the working directory

Reply:
Considering that the only difference between a Steam install and a non-Steam install is a launch option or two, I don't really see why the launcher should treat them so differently. Especially since the target directory would have to be treated correctly regardless.

I'd prefer if this would simply grab the registry option and store that in the launch options, if it turns out that we have to know if it's a Steam launch later then we can do that change later as well.

Reply:
alright, do you want to cast it to lower case in case someone only knows how to use capslock or cant type?
```
		if (tolower(arg.substr(0,9)) == "-registry")
                {
                    REGISTRY = " -registry:Steam";
                }
```
(i also changed registry_steam to just registry)

Reply:
I'd keep it simple, it's supposed to be fed by Steam after all, and I really hope that an automated software can spell correctly.

My suggestion would be;
```c++
if (tolower(arg.substr(0,10)) == "-registry:")
{
    registry_option = arg;
}
```

Let the launch concatenation take care of adding the space, as superfluous whitespace does absolutely no damage in launch argument lists, and I'd also suggest not using all uppercase names for non-const variables.

Reply:
alright, should be changed up now